tests/unittest/gcp_test.py

```diff
@@ -435,15 +436,14 @@ def testReadStartupScript(self):
     file = """#!/bin/bash
     echo 'This is a custom script'
     """
-    file_path = os.getcwd() + '/tmp.sh'
     try:
-      f = open(file_path, 'w')
-      f.write(file)
-      f.close()
-      os.environ['STARTUP_SCRIPT'] = file_path
-      script = FAKE_SOURCE_PROJECT._ReadStartupScript()
-      self.assertEqual(script, file)
-      os.remove(file_path)
+      with tempfile.NamedTemporaryFile() as temp:
+        temp.write(str.encode(file))
+        temp.flush()
+        os.environ['STARTUP_SCRIPT'] = temp.name
+        script = FAKE_SOURCE_PROJECT._ReadStartupScript()
+        self.assertEqual(script, file)
+        temp.close()
```

Comment:
No need to call `close()` if you're within a `with` statement, the cleanup is done automatically (even if an exception is raised)